### PR TITLE
Update Serving and Eventing to v0.20

### PR DESCRIPTION
**Pull request by mgencur**
Includes commits from the following PRs, and fixes upgrade tests.
https://github.com/openshift-knative/serverless-operator/pull/794
https://github.com/openshift-knative/serverless-operator/pull/791
https://github.com/openshift-knative/serverless-operator/pull/803

### CODE REVIEW

File: test/upgrade/upgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+func postUpgradeTests() []pkgupgrade.Operation {
+	tests := servingupgrade.ServingPostUpgradeTests()
+	return append(tests, eventingupgrade.PostUpgradeTest())
+}
+
+func continualTests() []pkgupgrade.BackgroundOperation {
+	tests := servingupgrade.ContinualTests()
+	return append(tests, eventingupgrade.ContinualTest())
```
Comment (markusthoemmes) [inline test/upgrade/upgrade_test.go:85]:
```suggestion
	return []pkgupgrade.BackgroundOperation{servingupgrade.ContinualTests()..., eventingupgrade.ContinualTest()...}
```

All other occasions of this pattern too. Makes it imo more clear, that we're running both eventing and serving.

File: test/upgrade/upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	eventingupgrade "knative.dev/eventing/test/upgrade"
+)
+
+func TestServerlessUpgrade(t *testing.T) {
+	c := newUpgradeConfig(t)
+	suite := pkgupgrade.Suite{
+		Tests: pkgupgrade.Tests{
+			PreUpgrade:    preUpgradeTests(),
+			PostUpgrade:   postUpgradeTests(),
+			Continual:     continualTests(),
```
Comment (markusthoemmes) [inline test/upgrade/upgrade_test.go:39]: Can we just inline the functions here? i.e.

```suggestion
			PostUpgrade:   []pkgupgrade.Operation{servingupgrade.ServingPostUpgradeTests()..., eventingupgrade.PostUpgradeTest()...},
			Continual:     []pkgupgrade.BackgroundOperation{servingupgrade.ContinualTests()..., eventingupgrade.ContinualTest()...},
```

File: test/upgrade/upgrade_test.go (hunk 1 above)
Comment (mgencur) [reply test/upgrade/upgrade_test.go:85]: You cannot do this in Golang :-D 
I can do this though:
```
var tests []pkgupgrade.Operation
tests = append(tests, servingupgrade.ServingPostUpgradeTests()...)
tests = append(tests, eventingupgrade.PostUpgradeTest())
return tests
```
I just thought the committed solution is clear enough and shorter. WDYT?

Comment (markusthoemmes) [reply test/upgrade/upgrade_test.go:85]: Nevermind then :joy:

File: test/upstream-e2e-tests.sh [hunk 3]
```diff
@@ -26,7 +26,7 @@ fi
 # Run upgrade tests
 if [[ $TEST_KNATIVE_UPGRADE == true ]]; then
   install_serverless_previous
-  run_rolling_upgrade_tests "${UPGRADE_TEST_SCOPE:-serving,eventing}"
```
Comment (cardil) [inline test/upstream-e2e-tests.sh:29]: Do we want to drop this flag? I personally haven't used it.

File: test/lib.bash [hunk 4]
```diff
@@ -179,145 +179,71 @@ function run_rolling_upgrade_tests {
...
 
-  if (( serving_in_scope )); then
-    end_serving_prober "${serving_prober_pid}"
-  fi
-  if (( eventing_in_scope )); then
-    end_eventing_prober "${eventing_prober_pid}"
-  fi
+  # Test configuration. See https://github.com/knative/eventing/tree/master/test/upgrade#probe-test-configuration
+  E2E_UPGRADE_TESTS_SERVING_SCALETOZERO=false \
+  E2E_UPGRADE_TESTS_SERVING_USE=true \
```
Comment (cardil) [inline test/lib.bash:204]:
```suggestion
  # Test configuration. See https://github.com/knative/eventing/tree/master/test/upgrade#probe-test-configuration
  # FIXME(ksuszyns): remove E2E_UPGRADE_TESTS_SERVING_SCALETOZERO when knative/operator#297 is fixed
  E2E_UPGRADE_TESTS_SERVING_SCALETOZERO=false \
  E2E_UPGRADE_TESTS_SERVING_USE=true \
```

**Review by cardil — CHANGES_REQUESTED** [review]

File: test/upstream-e2e-tests.sh (hunk 3 above)
Comment (mgencur) [reply test/upstream-e2e-tests.sh:29]: Yeah, I think we want to drop it. The variable is not defined anywhere else (e.g. vars.bash) so this change should be sufficient.

Comment (cardil) [reply test/upstream-e2e-tests.sh:29]: Okay.

**Comment by cardil** [comment]
Seams unrelated to changes: `test/servinge2e: TestNetworkPolicy`

/test 4.6-operator-e2e-aws-ocp-46
/lgtm

**Review by cardil — APPROVED** [review]
/lgtm

**Comment by cardil** [comment]
[`test/servinge2e: TestNetworkPolicy`](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/819/pull-ci-openshift-knative-serverless-operator-master-4.6-operator-e2e-aws-ocp-46/1360309846716452864) flakes again. See: #821 

/test 4.6-operator-e2e-aws-ocp-46

**Pull request merged**